src/main/java/com/gamingmesh/jobs/Jobs.java

```diff
@@ -997,6 +998,17 @@ public static void action(JobsPlayer jPlayer, ActionInfo info, Block block, Enti
 
 	    Boost boost = getPlayerManager().getFinalBonus(jPlayer, noneJob);
 
+        JobsPrePaymentEvent JobsPrePaymentEvent = new JobsPrePaymentEvent(jPlayer.getPlayer(), noneJob, income, pointAmount);
+        Bukkit.getServer().getPluginManager().callEvent(JobsPrePaymentEvent);
+        // If event is canceled, don't do anything
+        if (JobsPrePaymentEvent.isCancelled()) {
+            income = 0D;
+            pointAmount = 0D;
```

Comment:
Remove income, and pointAmount, you should use `return` variable, when the event was cancelled.

Reply:
The code mimics the behaviour of the JobsExpGainEvent, any reason this event should act differently?

Reply:
It doesn't mimic because the event class in this PR contains their other methods and more than in the JobsPaymentEvent or JobsExpGainEvent classes.

Reply:
```
		// JobsPayment event
		JobsExpGainEvent JobsExpGainEvent = new JobsExpGainEvent(jPlayer.getPlayer(), prog.getJob(), expAmount);
		Bukkit.getServer().getPluginManager().callEvent(JobsExpGainEvent);
		// If event is canceled, don't do anything
		if (JobsExpGainEvent.isCancelled())
		    expAmount = 0D;
		else
		    expAmount = JobsExpGainEvent.getExp();```
It very much does mimic the other event's behaviour

Reply:
Uh oh, sorry, forgot this event registration.